tools/lib/provision.py

```diff
@@ -431,7 +431,13 @@ def main(options):
         setup_node_modules(prefer_offline=True)
     except subprocess.CalledProcessError:
         print(WARNING + "`yarn install` failed; retrying..." + ENDC)
-        setup_node_modules()
+        try:
+            setup_node_modules()
+        except subprocess.CalledProcessError:
+            print(WARNING +
+                  "`yarn install` is failing, this usually indicates a problem with the network."
+                  + ENDC)
+            raise
```

Comment:
specifically, this line can be a `sys.exit(1)`.

Reply:
Cool, I'll make the changes.  This was actually my initial thought, but I didn't see a system exit elsewhere in `provision`, so that's why I went with `raise`.